cmd/virt-handler/virt-handler.go

```diff
@@ -158,6 +171,71 @@ func (app *virtHandlerApp) Run() {
 
 	factory := controller.NewKubeInformerFactory(virtCli.RestClient(), virtCli, namespace)
 
+	stop := make(chan struct{})
+	defer close(stop)
+
+	store, err := certificate.NewFileStore("kubevirt-client", app.CertDir, app.CertDir, "", "")
+	if err != nil {
+		glog.Fatalf("unable to initialize certificae store: %v", err)
+	}
+
+	certExpirationGauge := prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "virt_handler",
+			Subsystem: "certificate_manager",
+			Name:      "client_expiration_seconds",
+			Help:      "Gauge of the lifetime of a certificate. The value is the date the certificate will expire in seconds since January 1, 1970 UTC.",
+		},
+	)
+	prometheus.MustRegister(certExpirationGauge)
+
+	config := bootstrap.LoadCertConfigForNode(store, app.PodName, []string{app.PodName}, []net.IP{podIP})
+	config.CertificateExpiration = certExpirationGauge
+	manager, err := bootstrap.NewCertificateManager(config, virtCli.CertificatesV1beta1())
+	if err != nil {
+		glog.Fatalf("failed to request or fetch the certificate: %v", err)
+	}
+	go manager.Start()
+
+	certPool, err := certutil.NewPool("/var/run/secrets/kubernetes.io/serviceaccount/ca.crt")
+	if err != nil {
+		panic(err)
+	}
+
+	promTLSConfig := &tls.Config{
+		MinVersion: tls.VersionTLS12,
+		ClientCAs:  certPool,
+		RootCAs:    certPool,
+		GetCertificate: func(info *tls.ClientHelloInfo) (*tls.Certificate, error) {
+			cert := manager.Current()
+			if cert == nil {
+				return nil, fmt.Errorf("no serving certificate available for virt-handler")
+			}
+			return cert, nil
+		},
+	}
+
+	migrationTLSConfig := &tls.Config{
+		MinVersion: tls.VersionTLS12,
+		ClientAuth: tls.RequireAndVerifyClientCert,
```

Comment:
Maybe I'm super paranoid, but I want to make sure we have a test that exercises rejecting invalid client certs or at the very least verifies this option is always set. 

From a security standpoint, the ramifications of anything breaking in this security chain isn't good. whether it be a misconfigured TLS server, or a migration proxy that isn't properly utilizing tls.

Reply:
Definitely not paranoid. Needs a test.